### ISSUE TITLE
Check if the issue body is the same as the issue template

### ISSUE THREAD

**Issue by j-f1** [behaviorbot/request-info#8]
This app looks awesome!

Currently, it only checks if the body is empty:

https://github.com/behaviorbot/request-info/blob/68e08a896e0e838110e45b5263a042d6ff4bf260/index.js#L21

However, some repositories have a `.github/ISSUE_TEMPLATE.md` ([example](https://github.com/desktop/desktop/blob/master/.github/ISSUE_TEMPLATE.md)), which GitHub automatically puts in the issue entry box. Could this app check if the issue contents are the same as the template?

An even more awesome feature would be to check if there are any fields in the template that need to be filled out, then ask the user to fill them out.

**Comment by hiimbex** [comment]
Hi @j-f1, thanks for the feedback!

Issue templates was definitely something I considered when creating the bot. There is a [bot](https://github.com/kytrinyx/golang-tour-boilerplate-responder) built by @kytrinyx that closes issues that do not fill out the issue template. This app although specifically designed for one repo, could easily be made configurable and close issues that don't fill out any issue template fields. So this could definitely be done.

> An even more awesome feature would be to check if there are any fields in the template that need to be filled out, then ask the user to fill them out.

This could be a little more difficult to accomplish. The main problem I wouldn't be sure about would be determining what fields are missing? Most issue templates are just text, using your example, but a bit shortened:

```md
Description

[Description of the issue]

OS version: [version here]

Steps to Reproduce

1. [First step]
2. [Second step]
3. [and so on]
Expected behavior: [What you expected to happen]
```
How would the bot determine what fields are missing? If there is no text in between `Description` and `OS version`? In between each number? What if your use case only takes 2 steps to reproduce? In some issue templates, certain fields are optional, and I haven't seen a standardized way of displaying that. 

I think that @lee-dohm was actually interested in making a github app to solve this problem using some comparison of the original issue template to the body of the issue. Do you think `request-info` would be a good place for that to live?

Basically, I think this might be difficult to make work for many different issue templates, but I'd be +1 for adding the comment as well if the issue template wasn't edited at all.

**Comment by j-f1** [comment]
Maybe something like this:
```yaml
boilerplate:
  delimiters:
    - "["
    - "]"
  fields:
    - description of the issue
    - version here
    - first step
    - second step
    - and so on
    - what you expected to happen
```

It would search for `[description of the issue]`, `[version here]`, etc.

**Comment by hiimbex** [comment]
That could work, I was thinking before assuming we base this only off of information from the issue template. But where would the above live? In a `.github/config.yml` file? It's somewhat inconvenient because then what if you change your issue template and forget to update the config file?

**Comment by j-f1** [comment]
👍, let’s abandon the template-checking idea and just check for strict equality.

**Comment by hiimbex** [comment]
👍 I'll implement this shortly :)

**Comment by hiimbex** [comment]
So I didn't actually get around to this until now, but given the month of October approaching, I'm gonna open this up to any hacktoberfest hackers! :octocat: :robot: :octocat: 

The edits to code basically need to check that: 
1. if a `.github/ISSUE_TEMPLATE.md` exists, get it's contents, the code for that might look something like:
```js
const res = await context.github.repos.getContent(this.repo({path: '.github/ISSUE_TEMPLATE.md'}))
const issueTemplate = Buffer.from(res.data.content, 'base64').toString()
```
2. and it matches exactly the body of the issue opened
3. (There's a number of ways to do this, but I'll suggest a simple way) create a variable `issueTemplateCopied` and setting it to true if the `body` of the issue matches that of the `.github/ISSUE_TEMPLATE.md`.
4. Amend the following code:
https://github.com/behaviorbot/request-info/blob/d629c8a67b0ea13e2af818dad4435b79bf6e4865/index.js#L21
to be `if (!body || badTitle || issueTemplateCopied) {`

That should be it! Feel free to comment here if you have any questions.

**Comment by codebytere** [comment]
@hiimbex yo yo the electron team might be able to make some good use of this 😁 i'll take a swing at it soonish

**Comment by hiimbex** [comment]
@codebytere awesome! If you wanna try your hand at implementing some sort of checking to see if the issue template was actually filled out, feel free!

**Comment by j-f1** [comment]
@codebytere Have you had a chance to do this yet? If you’re busy, I can try to implement the feature this weekend.

**Comment by hiimbex** [comment]
I don't think anyone is desperate to implement this for hacktoberfest reasons, so whoever gets around to it first is fine with me! @j-f1 I'd say go for it! (Also if it's not implemented soonish, I can just do it too, I've just been busy lately)

**Comment by codebytere** [comment]
Yeah i can probably get to it this weekend since our need for it has increased recently

**Comment by szeck87** [comment]
I've created a bot similar to this, but checks for incomplete tasklists and keywords from the config file that should be in each issue. Not exactly looking at the issue template, but could go there eventually. https://github.com/szeck87/issue-complete

**Comment by rsarky** [comment]
Heyy, I would love to fix this one for you. :smile:

**Comment by szeck87** [comment]
Believe this was done with #32

**Comment by rsarky** [comment]
Thanks @szeck87

**Comment by hiimbex** [comment]
actually I think #32 applied to pull request templates, not issue templates. But there is already an open or for this issue.

**Comment by shiftkey** [comment]
I still think this is outstanding and requires considering a repository might define multiple templates in `.github/ISSUE_TEMPLATE`. I started on this in #39 and would love feedback to ensure I'm on the right track.
